samples/apps/autogen-studio/autogenstudio/web/websocketmanager.py

```diff
@@ -135,6 +64,35 @@ async def disconnect_all(self) -> None:
         for connection, _ in self.active_connections[:]:
             await self.disconnect(connection)
 
+    async def get_user_input(self, user_prompt: Dict, timeout: int, websocket: WebSocket) -> str:
+        await self.send_message(user_prompt, websocket)
+        # there's a bug I can't figure out.
+        # for some reason, the async with asyncio.timeout(..) context
+        # often forces you to wait until the timeout even when
+        # data = await websocket.receive_json() should have returned.
+        # so, as a janky workaround....
+        message = f"exit"
+        for i in range(timeout*10):
+            try:
+                data = await asyncio.wait_for(websocket.receive_json(), timeout=0.1)
+            except asyncio.TimeoutError:
+                continue
+            except WebSocketDisconnect:
+                print("Error: Tried to send a message to a closed WebSocket")
+                await self.disconnect(websocket)
+            except websockets.exceptions.ConnectionClosedOK:
+                print("Error: WebSocket connection closed normally")
+                await self.disconnect(websocket)
+            except Exception as e:
+                print(f"Error in sending message: {str(e)}", user_prompt)
+                await self.disconnect(websocket)
+
+            message = data.get("data").get("content")
```

Comment:
@SailorJoe6 

Thanks for the rewrite and `3.10` support. 

I am seeing some errors while testing ... .

With that fixed on my local version, I am still seeing a few other socket connection errors. 
Tried to address this  using a asynccontext manager but still seeing some of the timeout/delay issues where receive_json() does not return.


```
@asynccontextmanager
async def timeout_manager(timeout):
    try:
        yield await asyncio.wait_for(asyncio.sleep(timeout), timeout)
    except asyncio.TimeoutError:
        raise asyncio.TimeoutError

... 
async def get_user_input(self, user_prompt: Dict, timeout: int, websocket: WebSocket) -> str:
        await self.send_message(user_prompt, websocket)
        message = f"exit" 
        try:
            async with timeout_manager(20):
                try:
                    data = await websocket.receive_json()
                except WebSocketDisconnect:
                    print("Error: Tried to send a message to a closed WebSocket")
                    await self.disconnect(websocket)
                except websockets.exceptions.ConnectionClosedOK:
                    print("Error: WebSocket connection closed normally")
                    await self.disconnect(websocket)
                except Exception as e:
                    print(
                        f"Error in sending message: {str(e)}, {user_prompt}")
                    await self.disconnect(websocket)

                message = data.get("data").get("content")
                # break
        except asyncio.TimeoutError:
            print(">> Timeout")

        return message
```

Overall, once the core errors are addressed (even without the delay), I am happy to  mark this feature as experimental (always/terminate human input), merge in and then improve iteratively.  

Also, if you can attach a quick recording of how you are testing and results, that would be useful. Thanks.

Reply:
I'll address these issues to the best of my abilities tomorrow.  That is again for the review!

Reply:
Hey @victordibia, I'm working on these issues now.  I see there's a new commit on autogenstudio branch.  I can merge or rebase.  Personally, I prefer rebase, but it's your call.  Do you have a preference?